src/event-engine/states/handshaking.ts

```diff
@@ -24,25 +36,50 @@ HandshakingState.on(subscriptionChange.type, (context, event) => {
   return HandshakingState.with({ channels: event.payload.channels, groups: event.payload.groups });
 });
 
-HandshakingState.on(handshakingSuccess.type, (context, event) =>
-  ReceivingState.with({
-    channels: context.channels,
-    groups: context.groups,
-    cursor: event.payload,
-  }),
+HandshakingState.on(handshakeSuccess.type, (context, event) =>
+  ReceivingState.with(
+    {
+      channels: context.channels,
+      groups: context.groups,
+      cursor: {
+        timetoken: context.cursor?.timetoken ?? event.payload.timetoken,
+        region: event.payload.region,
+      },
+    },
+    [
+      emitStatus({
+        category: categoryConstants.PNConnectedCategory,
+      }),
+    ],
+  ),
 );
 
-HandshakingState.on(handshakingFailure.type, (context, event) =>
-  HandshakeReconnectingState.with({
-    ...context,
+HandshakingState.on(handshakeFailure.type, (context, event) => {
+  return HandshakeReconnectingState.with({
+    channels: context.channels,
+    groups: context.groups,
+    timetoken: context.cursor?.timetoken,
```

Comment:
All events in our specs use the `cursor` definition, which is both `timetoken` and `region` in one variable. Can we sync with specs in all places, where `timetoken` can be replaced with `cursor`? 
In this code, we're throwing off `region` but it is important in this case.

Reply:
unified ✔️